data/json/mapgen/dams.json

```
@@ -0,0 +1,233 @@
+        "  =_.......::....v.._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_..^............._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::..v...._=  ",
+        "  =_................_=  ",
+        "  =_.....^.::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  "
+      ],
+      "palettes": [ "dam_road_palette" ],
+      "items": { "_": { "item": "road", "chance": 1 } },
+      "vehicles": {
+        "^": { "vehicle": "highway", "chance": 10, "rotation": 90 },
+        "v": { "vehicle": "highway", "chance": 10, "rotation": 270 }
+      }
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "//": "Legacy dam that blocks river",
+    "om_terrain": [ "dam" ],
+    "object": {
+      "predecessor_mapgen": "river_shore",
+      "rows": [
+        "  =_................_=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::...v..._=  ",
+        "  =_.......::......._=  ",
+        "  =_...^............_=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_............v..._=  ",
+        "  =_.......::......._=  ",
+        "  =_.....^.::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_................_=  ",
+        "  =_.......::..v...._=  ",
+        "  =_.......::......._=  ",
+        "  =_..^....::......._=  ",
+        "  =_................_=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::....v.._=  ",
+        "  =_.......::......._=  ",
+        "  =_................_=  ",
+        "  =_....^..::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  "
+      ],
+      "palettes": [ "dam_ground_palette" ],
+      "items": { "_": { "item": "road", "chance": 1 } },
+      "vehicles": {
+        "^": { "vehicle": "highway", "chance": 10, "rotation": 90 },
+        "v": { "vehicle": "highway", "chance": 10, "rotation": 270 }
+      }
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "damhead_ground" ],
+    "object": {
+      "predecessor_mapgen": "river_shore",
+      "rows": [
+        "  =_................_=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_.......::......._=  ",
+        "  =_................_=  ",
+        "  =wuuuuuuuuuuuuuuuuw=  ",
+        "  =<UUUUUUUUUUUUUUUU<=  ",
+        "  ####################  ",
+        "  ####################  ",
+        "       ss######ss       ",
+        "        ss####ss        ",
+        "         ss##ss         ",
+        "          ssss          ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
+        "                        ",
```

```suggestion
        "          PPPP          ",
        "          PPPP          ",
        "          PPPP          ",
        "          PPPP          ",
        "          PPPP          ",
        "          PPPP          ",
        "          PPPP          ",
```
shouldn't these be filled as well?

---

Wasn't sure if we wanted to block all waterway access for boats